Integration/AndroidAAR/BabylonNative/src/main/java/BabylonNative/BabylonView.java

```diff
@@ -0,0 +1,113 @@
+package BabylonNative;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+import android.view.View;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class BabylonView extends SurfaceView implements SurfaceHolder.Callback2, View.OnTouchListener {
+    private static final String TAG = "BabylonView";
+    private boolean mViewReady = false;
+    private ViewDelegate mViewDelegate;
+
+    public BabylonView(Context context, ViewDelegate viewDelegate) {
+        super(context);
+        init(viewDelegate);
+        BabylonNative.Wrapper.initEngine(context.getResources().getAssets());
+    }
+
+    private void init(ViewDelegate viewDelegate) {
+        SurfaceHolder holder = getHolder();
+        holder.addCallback(this);
+        setOnTouchListener(this);
+        mViewDelegate = viewDelegate;
+    }
+
+    public void loadScript(String path) {
+        BabylonNative.Wrapper.loadScript(path);
+    }
+
+    public void eval(String source, String sourceURL) {
+        BabylonNative.Wrapper.eval(source, sourceURL);
+    }
+
+    public void onPause() {
+        setVisibility(View.GONE);
+        BabylonNative.Wrapper.activityOnPause();
+    }
+
+    public void onResume() {
+        BabylonNative.Wrapper.activityOnResume();
+    }
+
+    /**
+     * This method is part of the SurfaceHolder.Callback interface, and is
+     * not normally called or subclassed by clients of BabylonView.
+     */
+    public void surfaceCreated(SurfaceHolder holder) {
+        BabylonNative.Wrapper.surfaceCreated(getHolder().getSurface());
```

Comment:
How is the lifetime of the surface determined? Does the ANativeWindow pointer somehow keep it alive? Is there any thread synchronization we need to ensure the surface is not destroyed while the render loop is using it?

Reply:
Good point! I've put a mecanism to pause/resume rendering. I have to find a good way to test it.